test/test_rpc.py

```diff
@@ -129,17 +129,42 @@ def test_self_add(self):
         ):
             dist.rpc(self_worker_name, torch.add, args=(torch.ones(2, 2), 1))
 
-    def test_reinit(self):
+    @unittest.skipIf(
+        BACKEND != RpcBackend.PROCESS_GROUP,
+        "PROCESS_GROUP rpc backend specific test, skip"
+    )
+    def test_duplicate_name(self):
         store = dist.FileStore(self.file.name, self.world_size)
         dist.init_process_group(backend="gloo", rank=self.rank,
                                 world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel(self_name='worker{}'.format(self.rank),
+                                 backend=BACKEND,
+                                 self_rank=self.rank,
+                                 init_method=RPC_INIT_URL)
+        with self.assertRaisesRegex(RuntimeError,
+                                    "(is not unique|is already initialized)"):
             dist.init_model_parallel(self_name="duplicate_name",
                                      backend=BACKEND,
                                      self_rank=self.rank,
                                      init_method=RPC_INIT_URL)
```

Comment:
AFAICT this shouldn't initialize with `init_model_parallel` outside the `assertRaisesRegex` wrapper.

The purpose is to only test for duplicate names right? Not that it was already initialized.